File: Cap xA/Capitulo 07 - Exercicio 1a/c07ex1a.jl
#=
 * Livro .....: Algoritmos
 *              Logica para Desenvolvimento de Programacao de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 *
 * Capitulo ..: Apendice A - CORRECAO DE ALGUNS EXERCICIOS DE FIXACAO
 *
 * Programa ..: LISTA CLASSIFICADA POR NUMERO - DECRESCENTE
=#

using Printf

I = 0
J = 0
A = Vector{Int}(undef, 12)

println("CLASSIFICACAO (DECRESCENTE)")
println()

for I = 1:1:12
  @printf("Entre o %2do. numero: ", I)
  A[I] = parse(Int, readline())
end

for I = 1:1:11
  for J = I + 1:1:12
    if (A[I] < A[J])
      local X = A[I]
      A[I] = A[J]
      A[J] = X
    end
  end
end

println()
println("NUMEROS CLASSIFICADOS")
println()
for I = 1:1:12
  println(A[I])
end

File: Cap xA/Capitulo 07 - Exercicio 1a/c07ex1a.php
<?php
/**
 * Livro .....: Algoritmos
 *              Logica para Desenvolvimento de Programacao de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 * 
 * Capitulo ..: Apendice A - CORRECAO DE ALGUNS EXERCICIOS DE FIXACAO
 * 
 * Programa ..: LISTA CLASSIFICADA POR NUMERO - DECRESCENTE
 */

$A = array();

echo 'CLASSIFICACAO (DECRESCENTE)' . PHP_EOL;
echo PHP_EOL;

for ($I = 1; $I <= 12; $I++) {
    echo 'Entre o ' . str_pad($I, 2, ' ', STR_PAD_LEFT) . 'o. numero: ';
    $A[$I] = intval(readline());
}

for ($I = 1; $I <= 11; $I++) {
    for ($J = $I + 1; $J <= 12; $J++) {
        if ($A[$I] < $A[$J]) {
            $X = $A[$I];
            $A[$I] = $A[$J];
            $A[$J] = $X;
        }
    }
}

echo PHP_EOL;
echo 'NUMEROS CLASSIFICADOS' . PHP_EOL;
echo PHP_EOL;
for ($I = 1; $I <= 12; $I++) {
    echo $A[$I] . PHP_EOL;
}
?>
